lua/kznllm/init.lua

```diff
@@ -101,8 +101,6 @@ function M.get_visual_selection(opts)
   -- in visual block and visual line mode, we expect first column of srow and last column of erow
   if mode == 'V' or mode == '\22' or mode == 'n' then
     scol, ecol = 0, -1
-  else
-    ecol = ecol + 1
```

Comment:
I don't think removing this is correct. It is necessary for capturing the visual mode contents 
with `nvim_buf_get_text`

Just pulled this branch and the behavior I see is:

![review](https://github.com/user-attachments/assets/2ba1325c-ab2d-40eb-9d9c-889d92cf55e6)

Generally, the visual mode is inclusive of the current cursor position, so ecol + 1 is indeed required for this to work properly.

Maybe we need to make the case explicit to avoid confusion? `elseif mode == 'v`